code/modules/materials/Mat_Materials.dm

```diff
@@ -14,67 +14,264 @@
-	var/texture = ""
+	VAR_PROTECTED/texture = ""
 	/// How to blend the [/datum/material/var/texture].
-	var/texture_blend = BLEND_ADD
-
+	VAR_PROTECTED/texture_blend = BLEND_ADD
 	/// Should this even color the objects made from it? Mostly used for base station materials like steel
-	var/applyColor = 1
+	VAR_PROTECTED/applyColor = TRUE
 	/// The color of the material
-	var/color = "#FFFFFF"
+	VAR_PROTECTED/color = "#FFFFFF"
 	/// The "transparency" of the material. Kept as alpha for logical reasons. Displayed as percentage ingame.
-	var/alpha = 255
+	VAR_PROTECTED/alpha = 255
 	/// The 'quality' of the material
-	var/quality = 0
+	VAR_PROTECTED/quality = 0
 
 	/// The actual value of edibility. Changes internally and sets [/datum/material/var/edible].
-	var/edible_exact = 0
+	VAR_PROTECTED/edible_exact = 0
 	/// The functional value of edibility. Edible or not? This is what you check from the outside to see if material is edible. See [/datum/material/var/edible_exact].
-	var/edible = 0
+	VAR_PROTECTED/edible = 0
+
+	//triggers
+	/// Called when exposed to temperatures.
+	VAR_PROTECTED/list/triggersTemp = list()
+	/// Called when exposed to chemicals.
+	VAR_PROTECTED/list/triggersChem = list()
+	/// Called when owning object is picked up.
+	VAR_PROTECTED/list/triggersPickup = list()
+	/// Called when owning object is dropped.
+	VAR_PROTECTED/list/triggersDrop = list()
+	/// Called when exposed to explosions.
+	VAR_PROTECTED/list/triggersExp = list()
+	/// Called when the material is added to an object
+	VAR_PROTECTED/list/triggersOnAdd = list()
+	/// Called when the material is removed from an object
+	VAR_PROTECTED/list/triggersOnRemove = list()
+	/// Called when the life proc of a mob that has the owning item equipped runs.
+	VAR_PROTECTED/list/triggersOnLife = list()
+	/// Called when the owning object is used to attack something or someone.
+	VAR_PROTECTED/list/triggersOnAttack = list()
+	/// Called when a mob wearing the owning object is attacked.
+	VAR_PROTECTED/list/triggersOnAttacked = list()
+	/// Called when a mob wearing the owning object is shot.
+	VAR_PROTECTED/list/triggersOnBullet = list()
+	/// Called when *something* enters a turf with the material assigned. Also called on all objects on the turf with a material.
+	VAR_PROTECTED/list/triggersOnEntered = list()
+	/// Called when someone eats a thing with this material assigned.
+	VAR_PROTECTED/list/triggersOnEat = list()
+	/// Called when blob hits something with this material assigned.
+	VAR_PROTECTED/list/triggersOnBlobHit = list()
+	/// Called when an obj hits something with this material assigned.
+	VAR_PROTECTED/list/triggersOnHit = list()
 
-	var/owner_hasentered_added = FALSE
 
 	New()
 		. = ..()
 		for(var/datum/material_property/propPath as anything in concrete_typesof(/datum/material_property))
 			if(initial(propPath.default_value) > 0)
 				src.setProperty(initial(propPath.id), initial(propPath.default_value))
 
+	//getters for all the protected vars
```

Comment:
something something proc call overhead (probably fine?)

Reply:
anything querying material properties fast enough for that to be a problem should probably be rewritten anyway.